### PR TITLE
Adding deploying vm with GPU option

### PR DESCRIPTION
**Pull request by maayarosama**
### Description

Adding deploying vm with GPU option
### Changes

Adding gpu list to the vm
### Related Issues

#165

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstring

### CODE REVIEW

**Review by rawdaGastan — CHANGES_REQUESTED** [review]
we should update the docs

File: grid-cli/docs/vm.md
Comment (xmonader) [inline grid-cli/docs/vm.md]: alist -> a list, but i'm not sure how it will look still

File: grid-cli/cmd/deploy_vm.go
Comment (mariobassem) [inline grid-cli/cmd/deploy_vm.go:21]: i believe you need to upgrade zos version to have the GPU types

**Review by mariobassem — CHANGES_REQUESTED** [review]
i believe you need to update zos version

File: grid-cli/cmd/deploy_vm.go
Comment (rawdaGastan) [reply grid-cli/cmd/deploy_vm.go:21]: it will wait until client pr got merged

File: grid-cli/docs/vm.md
Comment (maayarosama) [reply grid-cli/docs/vm.md]: Should be something like this
```bash
tf-grid-cli deploy vm --name examplevm --ssh ~/.ssh/id_rsa.pub --cpu 2 --memory 4 --disk 10 --gpus gpu1 --gpus gpu2 --node 12```

**Review by rawdaGastan — APPROVED** [review]
tested 👍🏻

File: grid-cli/docs/vm.md
Comment (xmonader) [reply grid-cli/docs/vm.md]: is it the `id` string?

Comment (rawdaGastan) [reply grid-cli/docs/vm.md]: can be added to the docs (gpus IDs)?

**Review by xmonader — APPROVED** [review]

**Pull request merged**
